### PR TITLE
log image fix

### PR DESCRIPTION
**Pull request by jessechancy**
# What does this PR do?

Updates mlflow logger `log_image` to use the new API with time-dimension. This will enable viewing the images in MLflow

# What issue(s) does this change relate to?

# Before submitting
- [ ] Have you read the [contributor guidelines](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md)?
- [ ] Is this change a documentation change or typo fix? If so, skip the rest of this checklist.
- [ ] Was this change discussed/approved in a GitHub issue first? It is much more likely to be merged if so.
- [ ] Did you update any related docs and document your change?
- [ ] Did you update any related tests and add any new tests related to your change? (see [testing](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#running-tests))
- [ ] Did you run the tests locally to make sure they pass?
- [ ] Did you run `pre-commit` on your change? (see the `pre-commit` section of [prerequisites](https://github.com/mosaicml/composer/blob/dev/CONTRIBUTING.md#prerequisites))

### CODE REVIEW

**Review by mvpatel2000 — APPROVED** [review]

**Review by milocress — APPROVED** [review]
Approving to unblock. @mvpatel2000 can you take another look to make sure everything is shipshape? :ship: :ship: :ship:

File: composer/algorithms/utils/augmentation_primitives.py [hunk 1]
```diff
@@ -130,6 +152,12 @@ def equalize(pil_img: Image.Image, level: float):
     return ImageOps.equalize(pil_img)
 
 
+from PIL import WebPImagePlugin
+
+supported = WebPImagePlugin.SUPPORTED
+assert supported
```
Comment (milocress) [inline composer/algorithms/utils/augmentation_primitives.py:158]:
```suggestion
```

delete

File: composer/algorithms/utils/augmentation_primitives.py [hunk 2]
```diff
@@ -130,6 +152,12 @@ def equalize(pil_img: Image.Image, level: float):
     return ImageOps.equalize(pil_img)
 
 
+from PIL import WebPImagePlugin
+
+supported = WebPImagePlugin.SUPPORTED
```
Comment (milocress) [inline composer/algorithms/utils/augmentation_primitives.py:157]:
```suggestion
```

File: setup.py [hunk 3]
```diff
@@ -142,7 +142,7 @@ def package_files(prefix: str, directory: str, extension: str):
     'cryptography==42.0.8',
     'pytest-httpserver>=1.0.4,<1.1',
     'setuptools<=59.5.0',
-    'pillow==9.3.0',  # Matches the Pillow version listed in the Dockerfile
+    'pillow>=10.3.0',
```
Comment (mvpatel2000) [inline setup.py:145]:
```suggestion
    'pillow>=10.3.0,<11',
```
good to pin dependencies or things breaks

File: composer/algorithms/utils/augmentation_primitives.py [hunk 4]
```diff
@@ -225,7 +262,12 @@ def translate_y(pil_img: Image.Image, level: float):
     level = _int_parameter(_sample_level(level), pil_img.size[1] / 3)
     if np.random.random() > 0.5:
         level = -level
-    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, 0, 0, 1, level), resample=Image.BILINEAR)
+    return pil_img.transform(
+        pil_img.size,
+        Transform.AFFINE,  # pyright: ignore
+        (1, 0, 0, 0, 1, level),
+        resample=Resampling.BILINEAR,  # pyright: ignore
+    )  # pyright: ignore
```
Comment (mvpatel2000) [inline composer/algorithms/utils/augmentation_primitives.py:270]: why are there type ignores?

File: composer/algorithms/utils/augmentation_primitives.py [hunk 5]
```diff
@@ -211,7 +243,12 @@ def translate_x(pil_img: Image.Image, level: float):
     level = _int_parameter(_sample_level(level), pil_img.size[0] / 3)
     if np.random.random() > 0.5:
         level = -level
-    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, level, 0, 1, 0), resample=Image.BILINEAR)
+    return pil_img.transform(
+        pil_img.size,
+        Transform.AFFINE,  # pyright: ignore
+        (1, 0, level, 0, 1, 0),
+        resample=Resampling.BILINEAR,  # pyright: ignore
+    )  # pyright: ignore
```
Comment (mvpatel2000) [inline composer/algorithms/utils/augmentation_primitives.py:251]: same comment on type

File: composer/algorithms/utils/augmentation_primitives.py [hunk 6]
```diff
@@ -197,7 +224,12 @@ def shear_y(pil_img: Image.Image, level: float):
     level = _float_parameter(_sample_level(level), 0.3)
     if np.random.uniform() > 0.5:
         level = -level
-    return pil_img.transform(pil_img.size, Image.AFFINE, (1, 0, 0, level, 1, 0), resample=Image.BILINEAR)
+    return pil_img.transform(
+        pil_img.size,
+        Transform.AFFINE,  # pyright: ignore
+        (1, 0, 0, level, 1, 0),
+        resample=Resampling.BILINEAR,  # pyright: ignore
+    )  # pyright: ignore
```
Comment (mvpatel2000) [inline composer/algorithms/utils/augmentation_primitives.py:232]: same on type

File: composer/algorithms/utils/augmentation_primitives.py [hunk 7]
```diff
@@ -183,7 +205,12 @@ def shear_x(pil_img: Image.Image, level: float):
     level = _float_parameter(_sample_level(level), 0.3)
     if np.random.uniform() > 0.5:
         level = -level
-    return pil_img.transform(pil_img.size, Image.AFFINE, (1, level, 0, 0, 1, 0), resample=Image.BILINEAR)
+    return pil_img.transform(
+        pil_img.size,
+        Transform.AFFINE,  # pyright: ignore
+        (1, level, 0, 0, 1, 0),
+        resample=Resampling.BILINEAR,  # pyright: ignore
+    )  # pyright: ignore
```
Comment (mvpatel2000) [inline composer/algorithms/utils/augmentation_primitives.py:213]: why are there so many type ignore?

File: composer/algorithms/utils/augmentation_primitives.py [hunk 8]
```diff
@@ -155,7 +177,7 @@ def rotate(pil_img: Image.Image, level: float):
     degrees = _int_parameter(_sample_level(level), 30)
     if np.random.uniform() > 0.5:
         degrees = -degrees
-    return pil_img.rotate(degrees, resample=Image.BILINEAR)
+    return pil_img.rotate(degrees, resample=Resampling.BILINEAR)  # pyright: ignore
```
Comment (mvpatel2000) [inline composer/algorithms/utils/augmentation_primitives.py:180]: why type ignore?

File: composer/algorithms/utils/augmentation_primitives.py [hunk 9]
```diff
@@ -26,11 +26,33 @@
...
         * ``augmentation_sets["all"]`` contains all augmentations.
 """
+from enum import IntEnum
 from typing import Callable
 
 import numpy as np
 from PIL import Image, ImageEnhance, ImageOps
 
+try:
+    from PIL.Image import Resampling, Transform  # pyright: ignore
```
Comment (mvpatel2000) [inline composer/algorithms/utils/augmentation_primitives.py:36]: why is type ignore needed?

File: composer/algorithms/utils/augmentation_primitives.py [hunk 10]
```diff
@@ -26,11 +26,33 @@
...
+except ImportError:  # fix version conflict issues
```
Comment (mvpatel2000) [inline composer/algorithms/utils/augmentation_primitives.py:37]: why is this needed?

Comment (milocress) [reply composer/algorithms/utils/augmentation_primitives.py]: Update: am goose, not needed

File: composer/algorithms/utils/augmentation_primitives.py (hunk 9 above)
Comment (milocress) [reply composer/algorithms/utils/augmentation_primitives.py]: Update: am goose, not needed

File: composer/algorithms/utils/augmentation_primitives.py (hunk 8 above)
Comment (milocress) [reply composer/algorithms/utils/augmentation_primitives.py]: Update: am goose, not needed

File: composer/algorithms/utils/augmentation_primitives.py (hunk 6 above)
Comment (milocress) [reply composer/algorithms/utils/augmentation_primitives.py]: Update: am goose, not needed

File: composer/algorithms/utils/augmentation_primitives.py (hunk 5 above)
Comment (milocress) [reply composer/algorithms/utils/augmentation_primitives.py]: Update: am goose, not needed

File: composer/algorithms/utils/augmentation_primitives.py (hunk 4 above)
Comment (milocress) [reply composer/algorithms/utils/augmentation_primitives.py]: Update: am goose, not needed

**Review by mvpatel2000 — APPROVED** [review]
LGTM!
